src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -326,4 +326,7 @@ public static int GetTrailingWhitespaceLength(this SyntaxNode node, FormattingCo
 
         return whitespaceLength;
     }
+
+    public static SyntaxNode? FindNode(this SyntaxNode node, int index)
```

Comment:
I'm a little leary of this method. A single index in source is going to have a number of nodes associated with it, from the RazorDocumentSyntax all the way down. Consider naming it something like `FindInnermostNode`, or documenting it's behavior.

Reply:
For reference, the similar method in roslyn takes a span, not a single location, and has an `getInnermostNodeForTie` parameter. You can't really tie without a span, but the innermost/outermost concept is still important.

Reply:
I'd be fine with `FindInnermostNode` make sense to me, but at the same time, I don't think anyone could expect this to find the outermost node, or it would always find the root.

Reply:
I updated to `FindInnermostNode` for clarity, since I think we'll need to add a `FindNode` with a span like Roslyn has